mypy/messages.py

```diff
@@ -1095,11 +1096,17 @@ def need_annotation_for_var(self, node: SymbolNode, context: Context,
             type_dec = '<type>'
             if alias == 'Dict':
                 type_dec = '{}, {}'.format(type_dec, type_dec)
-            if python_version < (3, 6):
-                hint = ' (hint: "{} = ...  # type: {}[{}]")'.format(node.name(), alias, type_dec)
-            else:
+            if has_variable_annotations:
                 hint = ' (hint: "{}: {}[{}] = ...")'.format(node.name(), alias, type_dec)
-        self.fail("Need type annotation for '{}'{}".format(unmangle(node.name()), hint), context)
+            else:
+                hint = ' (hint: "{} = ...  # type: {}[{}]")'.format(node.name(), alias, type_dec)
+
+        if has_variable_annotations:
+            needed = 'annotation'
+        else:
+            needed = 'comment'
+
+        self.fail("Need type {} for '{}'{}".format(needed, unmangle(node.name()), hint), context)
```

Comment:
Note: this make it impossible to grep for "Need type annotation" in the source to find this.

Reply:
"impossible"? not at all! You can grep for "Need type .* for" or even pass -e with multiple expressions or do grep -E with a `|` in the expression.

Reply:
Yes, I've found it after all.. ;)
This was just meant as feedback that this pattern makes it harder to grep for (log) messages.

Reply:
That is fair and well received feedback, but I think that will get solved by Jukka's error code work anyway. Thank you for raising this point all the same.

Reply:
Ok, sorry for the exaggeration.
My main question is the one below, i.e. if this should be a note instead.